cmd/subnetcmd/upgradecmd/vm_test.go

```diff
@@ -274,3 +282,53 @@ func TestAtMostOneAutomationSelected(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdateToCustomBin(t *testing.T) {
+	assert := require.New(t)
+	testDir := t.TempDir()
+
+	subnetName := "testSubnet"
+	sc := models.Sidecar{
+		Name:       subnetName,
+		VM:         models.SubnetEvm,
+		VMVersion:  "v3.0.0",
+		RPCVersion: 20,
+		Subnet:     subnetName,
+	}
+	networkToUpgrade := futureDeployment
+
+	factory := logging.NewFactory(logging.Config{})
+	log, err := factory.Make("avalanche")
+	assert.NoError(err)
+
+	// create the user facing logger as a global var
+	ux.NewUserLog(log, os.Stdout)
+
+	app = &application.Avalanche{}
+	app.Setup(testDir, log, config.New(), prompts.NewPrompter(), application.NewDownloader())
+
+	err = os.MkdirAll(app.GetSubnetDir(), constants.DefaultPerms755)
+	assert.NoError(err)
+
+	err = app.CreateSidecar(&sc)
+	assert.NoError(err)
+
+	err = os.MkdirAll(app.GetCustomVMDir(), constants.DefaultPerms755)
+	assert.NoError(err)
+
+	newBinary = "../../../README.md"
```

Comment:
Also, generally it's not a good idea to use global variables in tests (and generally always try to avoid globals when possible)

Reply:
We haven't seen this pattern in tests a bunch because we haven't actually tested most of the functions that rely on command line arguments with unit tests

Reply:
I switched to a specific asset